### PR TITLE
fix WALInputStream not closed

### PR DESCRIPTION
**Pull request by shuwenwei**
## Description
fix WALInputStream not closed

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALByteBufReader.java [hunk 1]
```diff
@@ -20,38 +20,46 @@
...
-  private final WALMetaData metaData;
-  private final DataInputStream logStream;
-  private final Iterator<Integer> sizeIterator;
+  private WALMetaData metaData;
+  private DataInputStream logStream;
+  private Iterator<Integer> sizeIterator;
 
   public WALByteBufReader(File logFile) throws IOException {
-    this(logFile, FileChannel.open(logFile.toPath(), StandardOpenOption.READ));
+    WALInputStream walInputStream = new WALInputStream(logFile);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALByteBufReader.java:42]: try-with-resource?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALByteBufReader.java [hunk 2]
```diff
@@ -20,38 +20,46 @@
...
 
-  public WALByteBufReader(File logFile, FileChannel channel) throws IOException {
-    this.logFile = logFile;
-    this.channel = channel;
-    this.logStream = new DataInputStream(new WALInputStream(logFile));
-    this.metaData = WALMetaData.readFromWALFile(logFile, channel);
-    this.sizeIterator = metaData.getBuffersSize().iterator();
-    channel.position(0);
+  public WALByteBufReader(WALEntryPosition walEntryPosition) throws IOException {
+    WALInputStream walInputStream = walEntryPosition.openReadFileStream();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALByteBufReader.java:54]: same as above

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java [hunk 3]
```diff
@@ -60,10 +60,15 @@ public class WALInputStream extends InputStream implements AutoCloseable {
...
-    fileSize = channel.size();
     this.logFile = logFile;
-    analyzeFileVersion();
-    getEndOffset();
+    try {
+      fileSize = channel.size();
+      analyzeFileVersion();
+      getEndOffset();
+    } catch (Exception e) {
+      channel.close();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java:69]: if we use try-with-resource for all it's caller or caller's caller, maybe we do not need this ?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Review by jt2594838 — APPROVED** [review]

**Pull request merged**